platform/vcs-impl/src/com/intellij/openapi/vcs/actions/VcsContextWrapper.java

```diff
@@ -160,7 +160,7 @@ public Refreshable getRefreshableDialog() {
       stream(VcsDataKeys.FILE_PATH_ARRAY.getData(myContext)),
       getSelectedFilesStream().map(VcsUtil::getFilePath),
       stream(getSelectedIOFiles()).map(VcsUtil::getFilePath)
-    );
+    ).distinct();
```

Comment:
While 'distinct' does indeed process all elements it is technically not a terminal operation, that is it does not process all elements until a terminal operation is invoked.  However this change is not really critical. If you prefer we can move 'distinct' to 'getSelectedFilePaths' or even remove the change altogether.